sbpy/thermal/core.py

```diff
@@ -5,66 +5,274 @@
 
+import abc
+import numpy as np
+from scipy.integrate import dblquad
+import astropy.units as u
+import astropy.constants as const
+from astropy.modeling.models import BlackBody
+from ..data import (Phys, Obs, Ephem, dataclass_input,
+                    quantity_to_dataclass)
+from ..vector import twovec, xyz2sph, sph2xyz
 
-class ThermalClass():
 
-    def flux(phys, eph, lam):
-        """Model flux density for a given wavelength `lam`, or a list/array thereof
+class ThermalClass(abc.ABC):
+    """Abstract base class for thermal models.
+
+    This class implements the basic calculation for thermal models,
+    such as integration of total flux based on a temperature distribution.
+    """
+
+    @u.quantity_input(rh=u.km, R=u.km)
+    def __init__(self, rh, R, albedo=0.1, emissivity=1., beaming=1.):
+        """
+        Parameters
+        ----------
+        rh : u.Quantity
+            Heliocentric distance
+        R : u.Quantity
+            Radius of asteroid
+        albedo : float, u.Quantity
+            Bolometric Bond albedo
+        emissivity : float, u.Quantity
+            Emissivity of surface
+        beaming : float, u.Quantity
+            Beaming parameter
+        """
+        self.rh = rh
+        self.R = R
+        self.albedo = albedo
+        self.emissivity = emissivity
+        self.beaming = beaming
+
+    @abc.abstractmethod
+    def T(self, lon, lat):
+        """Temperature on the surface of an object.
+
+        Needs to be overridden in subclasses.  This function needs to be able
+        to return a valid quantity for the full range of lon and lat, i.e.,
+        include the night side of an object.
+
+        lon : u.Quantity
+            Longitude
+        lat : u.Quantity
+            Latitude
+        """
+        pass
+
+    @u.quantity_input(wave_freq=u.m, equivalencies=u.spectral())
+    def _int_func(self, lon, lat, m, unit, wave_freq):
+        """Integral function for `fluxd`.
 
         Parameters
         ----------
-        phys : `sbpy.data.Phys` instance, mandatory
-            provide physical properties
-        eph : `sbpy.data.Ephem` instance, mandatory
-            provide object ephemerides
-        lam : `astropy.units` quantity or list-like, mandatory
-            wavelength or list thereof
-
-        Examples
-        --------
-        >>> from astropy.time import Time
-        >>> from astropy import units as u
-        >>> from sbpy.thermal import STM
-        >>> from sbpy.data import Ephem, Phys
-        >>> epoch = Time('2019-03-12 12:30:00', scale='utc')
-        >>> eph = Ephem.from_horizons('2015 HW', location='568', epochs=epoch) # doctest: +REMOTE_DATA
-        >>> phys = PhysProp('diam'=0.3*u.km, 'pv'=0.3) # doctest: +SKIP
-        >>> lam = np.arange(1, 20, 5)*u.micron # doctest: +SKIP
-        >>> flux = STM.flux(phys, eph, lam) # doctest: +SKIP
-
-        not yet implemented
+        lon : float
+            Longitude in radiance
```

Comment:
What does this mean, longitude/latitude in "radiance"?

Reply:
Ah, typo.  Should be radian.